onnxruntime/core/providers/nuphar/scripts/symbolic_shape_infer.py

```diff
@@ -39,67 +39,75 @@ def get_opset(mp, domain=['', 'onnx']):
             return opset.version
     return None
 
+def as_scalar(x):
+    if type(x) == list:
+        assert len(x) == 1
+        return x[0]
+    elif type(x) == np.ndarray:
+        return np.asscalar(x)
+    else:
+        return x
+
 class SymbolicShapeInference:
     def __init__(self, auto_merge, verbose):
         self.dispatcher_ = {
+            'AveragePool'       : self._infer_Pool,
             'Cast'              : self._infer_Cast,
             'CategoryMapper'    : self._infer_CategoryMapper,
             'Compress'          : self._infer_Compress,
             'Concat'            : self._infer_Concat,
+            'Conv'              : self._infer_Conv,
```

Comment:
Not sure if we try to maintain alphabetic order or not. If we do, Conv might need to be moved below ConstantOfShape.

Reply:
Sure

---
In reply to: [322352071](https://github.com/microsoft/onnxruntime/pull/1787#discussion_r322352071) [](ancestors = 322352071)